StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerDataSource.swift

```diff
@@ -23,12 +23,14 @@ protocol LinkAccountPickerDataSource: AnyObject {
     var nextPaneOnAddAccount: FinancialConnectionsSessionManifest.NextPane? { get set }
     var analyticsClient: FinancialConnectionsAnalyticsClient { get }
     var dataAccessNotice: FinancialConnectionsDataAccessNotice? { get }
+    var acquireConsentOnPrimaryCtaClick: Bool { get }
 
     func updateSelectedAccounts(_ selectedAccounts: [FinancialConnectionsAccountTuple])
     func fetchNetworkedAccounts() -> Future<FinancialConnectionsNetworkedAccountsResponse>
     func selectNetworkedAccounts(
         _ selectedAccounts: [FinancialConnectionsPartnerAccount]
     ) -> Future<FinancialConnectionsInstitutionList>
+    func markConsentAcquired() -> Promise<FinancialConnectionsSessionManifest>
```

Comment:
Could you help me understand why this API returns a `Promise`, while the others return a `Future`?

Reply:
@mats-stripe 

this was just a copy-paste that's faulty from old code (consent data source)

it should be a Future

_(there's a longer backstory where the Future/Promise implementation is questionable)_

Reply:
fixed here, thank you! https://github.com/stripe/stripe-ios/pull/3870/commits/7c988078c4e02fe0a2ac124c22f51914025cec00